Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/EditorLayerComponent.cpp

```diff
@@ -67,6 +67,14 @@ namespace AzToolsFramework
             }
         }
 
+        EditorLayer::~EditorLayer()
+        {
+            for (AZ::Entity* entity : m_layerEntities)
```

Comment:
We can change the `m_layerEntities` type to `AZStd;:vector<AZStd::unique_ptr<AZ::Entity>>`